### PR TITLE
support unWritable props in the logger option

### PR DESCRIPTION
**Pull request by zhangwinning**
Fixs https://github.com/fastify/fastify/issues/2455

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/logger.test.js [hunk 1]
```diff
@@ -1480,3 +1480,15 @@ test('should not log incoming request and outgoing response when disabled', t =>
...
   })
 })
+
+test('should pass when using unWritable props in the logger option', t => {
+  t.plan(1)
+  try {
+    Fastify({
+      logger: Object.defineProperty({}, 'level', { value: 'info' })
+    })
+    t.pass()
```
Comment (RafaelGSS) [inline test/logger.test.js:1490]: What do you think to test if this works with: `Object.defineProperties({}, { level: 'info', stream: streamExample })` like the first test suite?

Comment (zhangwinning) [reply test/logger.test.js:1490]: The above test case check object assignment value when the  object's properties unWritable, I think the 'level' or 'stream' is same

File: test/logger.test.js [hunk 2]
```diff
@@ -1480,3 +1480,15 @@ test('should not log incoming request and outgoing response when disabled', t =>
...
+  } catch (e) {
+    t.fail()
+  }
```
Comment (L2jLiga) [inline test/logger.test.js:1493]: Let's avoid useless catch, see https://github.com/fastify/fast-json-stringify/issues/229

File: lib/logger.js [hunk 3]
```diff
@@ -81,9 +81,20 @@ function createLogger (options) {
     logger.child = () => logger
     return { logger, hasLogger: false }
   } else {
-    options.logger = typeof options.logger === 'object' ? options.logger : {}
-    options.logger.level = options.logger.level || 'info'
-    options.logger.serializers = Object.assign({}, serializers, options.logger.serializers)
+    const ops = {}
+    if (typeof options.logger === 'object') {
```
Comment (jsumners) [inline lib/logger.js:85]:
```suggestion
    if (Object.prototype.toString.call(options.logger) === '[object Object]') {
```

File: lib/logger.js [hunk 4]
```diff
@@ -81,9 +81,20 @@ function createLogger (options) {
     logger.child = () => logger
     return { logger, hasLogger: false }
   } else {
-    options.logger = typeof options.logger === 'object' ? options.logger : {}
-    options.logger.level = options.logger.level || 'info'
-    options.logger.serializers = Object.assign({}, serializers, options.logger.serializers)
+    const ops = {}
```
Comment (jsumners) [inline lib/logger.js:84]:
```suggestion
    const localLoggerOptions = {}
```

^ this is clearer than having to look back to see what `ops` is.

File: test/logger.test.js (hunk 2 above)
Comment (zhangwinning) [reply test/logger.test.js:1493]: thanks, got it

File: lib/logger.js (hunk 3 above)
Comment (zhangwinning) [reply lib/logger.js:85]: Hi, thanks you suggestion😁，while I think It is more concise to use typeof whether it is an object, I want know using below benefit ？ 
```
Object.prototype.toString.call(options.logger) === '[object Object]'
```

File: lib/logger.js (hunk 4 above)
Comment (zhangwinning) [reply lib/logger.js:84]: thanks, I got it

File: lib/logger.js (hunk 3 above)
Comment (jsumners) [reply lib/logger.js:85]: `typeof [] === 'object'` is `true`

Comment (jsumners) [reply lib/logger.js:85]: `typeof null === 'object'` is `true`

Comment (zhangwinning) [reply lib/logger.js:85]: thanks

**Pull request merged**